pkg/controller/status/status.go

```diff
@@ -129,8 +132,31 @@ func (c *Controller) merge(existing *configv1.ClusterOperator) *configv1.Cluster
 			klog.Errorf("Programmer error: status source %d %T reported an empty message: %#v", i, source, summary)
 			continue
 		}
-		reason = summary.Reason
-		errors = append(errors, summary.Message)
+
+		degradingFailure := true
+
+		if summary.Operation == controllerstatus.Uploading {
+			if summary.Count < uploadFailuresCountThreshold {
+				klog.V(4).Infof("Number of last upload failures %d lower than threshold %d. Not marking as degraded.", summary.Count, uploadFailuresCountThreshold)
+				degradingFailure = false
+			} else {
+				klog.V(4).Infof("Number of last upload failures %d exceeded than threshold %d. Marking as degraded.", summary.Count, uploadFailuresCountThreshold)
+				degradingFailure = true
```

Comment:
nit: not needed

Reply:
I wanted to be a bit explicit here, but can do both ways.